File: leetcode/AIcoding/1.Gemini-2.5/java_solutions/973-k-closest-points-to-origin.java
import java.util.PriorityQueue;
import java.util.Comparator;

class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>(
            new Comparator<int[]>() {
                @Override
                public int compare(int[] p1, int[] p2) {
                    return (p2[0] * p2[0] + p2[1] * p2[1]) - (p1[0] * p1[0] + p1[1] * p1[1]);
                }
            }
        );

        for (int[] point : points) {
            pq.offer(point);
            if (pq.size() > k) {
                pq.poll();
            }
        }

        int[][] result = new int[k][2];
        for (int i = 0; i < k; i++) {
            result[i] = pq.poll();
        }

        return result;
    }
}

File: leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/973-k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        auto compare = [](const vector<int>& a, const vector<int>& b) {
            return (a[0] * a[0] + a[1] * a[1]) > (b[0] * b[0] + b[1] * b[1]);
        };

        priority_queue<vector<int>, vector<vector<int>>, decltype(compare)> pq(compare);

        for (const auto& point : points) {
            pq.push(point);
        }

        vector<vector<int>> result;
        for (int i = 0; i < k; ++i) {
            result.push_back(pq.top());
            pq.pop();
        }

        return result;
    }
};